supervision/metrics/detection.py

```diff
@@ -0,0 +1,326 @@
+from dataclasses import dataclass
+from typing import Callable, List, Optional, Tuple
+import matplotlib
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import supervision as sv
+from supervision.detection.utils import box_iou_batch
+
+
+@dataclass
+class ConfusionMatrix:
+    """
+    Data class containing information about classification results in form of a confusion matrix.
+    Attributes:
+        matrix: An array of shape (len(classes) + 1, len(classes) + 1) containing the number of TP, FP, FN and TN for each class.
+        classes: all known class names.
+        conf_threshold: detection confidence threshold between 0 and 1. Detections with lower confidence will be excluded from the matrix.
+        iou_threshold: detection iou threshold between 0 and 1. Detections with lower iou will be classified as FP.
+    """
+
+    matrix: np.ndarray
+    classes: List[str]
+    conf_threshold: float
+    iou_threshold: float
+
+    @classmethod
+    def from_matrix(
```

Comment:
I'm not sure we need that method. If you think about it it is a very thin layer over `dataclass` constructor. I'd recommend dropping this class method.

Reply:
At this point, it is redundant indeed. Removing until good times

Reply:
Awesome, let's keep it simple. ;)